### PR TITLE
Basic Cargo based build

### PR DESCRIPTION
**Pull request by ebfe**
Builds the uutils multicall binary containing all utils (except stdbuf)
by default. To only build a subset
    `cargo --no-default-features --features "<utils>"`
can be used.

Whats missing is building the standalone binaries and a mechanism to
automatically disable the build of unix only utils on windows.

Maybe a step towards fixing #655.

### CODE REVIEW

**Comment by Heather** [comment]
should we merge it?

**Comment by ebfe** [comment]
There was some resistance last time I tried to get a cargo build merged.
cc @Arcterus @jbcrail

**Comment by ebfe** [comment]
ping

**Comment by hexsel** [comment]
What were the objections? The Makefile approach to dependencies is not really that great.

**Comment by ebfe** [comment]
See #395

**Comment by hexsel** [comment]
Hm I kind of like having more granular configuration even if there are a few lines that are superfluous. Having said that, it's not like I actually contribute much. 

The make build seems to take a lot longer than your cargo build, even under --release.  Are you building everything `make` is? it's like ~1m (30s without --release) versus ~4m for make.  If you are, this would be a reason enough to upgrade.

File: Cargo.toml [hunk 1]
```diff
@@ -1,268 +1,162 @@
...
+uptime   = { optional=true, path="src/uptime" }
+users    = { optional=true, path="src/users" }
+wc       = { optional=true, path="src/wc" }
+whoami   = { optional=true, path="src/whoami" }
+yes      = { optional=true, path="src/yes" }
+
+
+[[bin]]
+name="uutils"
+path="src/uutils/main.rs"
```
Comment (hexsel) [inline Cargo.toml]: this needs to be 

```
path="src/uutils/uutils_cargo.rs"
```

Comment (ebfe) [inline Cargo.toml]: Oops, thanks. Not sure how I missed this. Fixed.

**Comment by thomanski** [comment]
It'd certainly be helpful to have a cargo build to run things like [clippy](https://github.com/Manishearth/rust-clippy) and would generally be more welcoming to newbies (such as myself) who haven't got much of a background in C/C++ and thus are not well-versed in make.

I could imagine it'd end up benefiting the project despite the extra effort of having to maintain both cargo and make builds.

**Comment by Arcterus** [comment]
I suppose we could merge this and see which system we prefer (I assume we would remove one of them at some point).

**Comment by Heather** [comment]
Then I'm merging this. If there will be serious problems please revert

**Pull request merged**